packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -321,11 +323,20 @@ class IOSDevice extends Device {
       return LaunchResult.failed();
     }
 
+    // Step 2.5: Generate a potential open port using the provided argument,
+    // or randomly with the package name as a seed.
+    final int assumedObservatoryPort = debuggingOptions?.deviceVmServicePort
+      ?? math.Random(packageId.hashCode).nextInt(1000) + 6000;
```

Comment:
We should probably pick a random port in the standard ephemeral port range 49152 to 65535 (2^15+2^14 to 2^16−1)

https://en.wikipedia.org/wiki/Ephemeral_port